app/views/ads/_ad.html.erb

```diff
@@ -1,61 +1,73 @@
-        <div class="control">
-          <div class="tags has-addons">
-            <%= category_tag ad %>
-          </div>
-        </div>
+<div class="tile is-parent mb-2 is-12">
+  <div class="tile is-child card is-flex is-flex-direction-column">
+    <div class="card-content" data-controller="card-dropdown">
 
-        <div class="control">
-          <div class="tags has-addons">
-            <span class="tag">Grad</span>
-            <span class="tag is-info"><%= ad.zip_and_city%></span>
+      <div class="is-flex is-justify-content-space-between mb-4">
+        <%= category_tag ad %>
+
+        <div class="dropdown is-right ml-2" data-card-dropdown-target="dropdown">
+          <div class="dropdown-trigger">
+            <button class="button" aria-haspopup="true" aria-controls="dropdown-menu-card-actions" data-action="card-dropdown#toggle">
+              <span class="icon">
+                <%= inline_svg_tag('dots-vertical.svg', aria: true) %>
+              </span>
+            </button>
+          </div>
+          <div class="dropdown-menu" id="dropdown-menu-card-actions" role="menu">
+            <div class="dropdown-content" data-controller="clipboard">
+              <a href="<%= ad_path(ad.to_param) %>" class="dropdown-item is-flex is-align-items-center">
+                <span class="icon is-small mr-2">
+                  <%= inline_svg_tag('link.svg', aria: true) %>
+                </span>
+                Otvori oglas
+              </a>
+              <a class="dropdown-item is-flex is-align-items-center" data-clipboard-target="button" data-clipboard-text="<%= ad_url(ad.to_param) %>">
+                <span class="icon is-small mr-2">
+                  <%= inline_svg_tag('clipboard.svg', aria: true) %>
+                </span>
+                Kopiraj poveznicu
+              </a>
+            </div>
           </div>
         </div>
+      </div>
+
+      <div class="content">
+        <h3 class="is-size-5"><%= ad.zip_and_city %></h3>
+
+        <%= simple_format ad.description %>
 
         <% if ad.address.present? %>
-          <div class="control">
-            <div class="tags has-addons">
-              <span class="tag">Adresa</span>
-              <span class="tag is-info"><%= ad.address %></span>
+          <div>
+            <div class="is-flex">
+              <span class="icon mr-2">
+                <%= inline_svg_tag('location.svg', aria: true) %>
+              </span>
+              <%= ad.address %>
             </div>
-          </div>
-        <% end %>
 
-        <div class="control">
-          <a href="<%= maps_url(ad.full_address) %>" target="_blank" class="is-block">
-            <div class="tags has-addons">
-              <span class="tag">Lokacija na karti</span>
-              <span class="tag is-info">
-                <span class="icon is-small">
+            <div class="mt-2">
+              <a href="<%= maps_url(ad.full_address) %>" target="_blank" class="is-flex">
```

Comment:
We could make the whole address a link perhaps?

I'd also duplicate the city name and zip here. Otherwise it's a bit confusing to see something like:

`14th Ave`
`Location on map`
🤔

Reply:
Totally agree, I found it redundant as well.